### ISSUE TITLE
Add support for Senlin clustering

### ISSUE THREAD

**Issue by JackKuei** [gophercloud/gophercloud#823]
Currently there is no support for senlin clustering in gophercloud.

[https://developer.openstack.org/api-ref/clustering/](url)
---
- [ ] Profiles: List
- [ ] Profiles: Create
- [ ] Profiles: Get
- [ ] Profiles: Update
- [ ] Profiles: Delete
- [ ] Profiles: Validate
---
- [ ] PolicyTypes: List
- [ ] PolicyTypes: Get
---
- [ ] Policies: Get
- [ ] Policies: Create
- [ ] Policies: Get
- [ ] Policies: Update
- [ ] Policies: Delete
- [ ] Policies: Validate
---
- [ ] Clusters: List
- [ ] Clusters: Create
- [ ] Clusters: Get
- [ ] Clusters: Update
- [ ] Clusters: Delete
- [ ] Clusters: Resize
- [ ] Clusters: ScaleIn
- [ ] Clusters: ScaleOut
- [ ] Clusters: AddNode
- [ ] Clusters: RemoveNode
- [ ] Clusters: ReplaceNode
- [ ] Clusters: AttachPolicy
- [ ] Clusters: DetachPolicy
- [ ] Clusters: UpdatePolicy
- [ ] Clusters: CollectAttributes
- [ ] Clusters: CheckHealth
- [ ] Clusters: Recover
- [ ] Clusters: Ops
---
- [ ] ClusterPolicies: List
- [ ] ClusterPolicies: Get
---
- [ ] Nodes: List
- [ ] Nodes: Create
- [ ] Nodes: Adopt
- [ ] Nodes: AdoptPreview
- [ ] Nodes: Get
- [ ] Nodes: Update
- [ ] Nodes: Delete
- [ ] Nodes: CheckHealth
- [ ] Nodes: Recover
- [ ] Nodes: Ops
---
- [ ] Receivers: List
- [ ] Receivers: Create
- [ ] Receivers: Get
- [ ] Receivers: Update
- [ ] Receivers: Delete
- [ ] Receivers: Notify
---
- [ ] Events: List
- [ ] Events: Get
---
- [ ] Webhooks: Trigger
---
- [ ] Actions:  List
- [ ] Actions:  Get
---
- [ ] Services:  List

**Comment by jtopjian** [comment]
@JackKuei I've lost track of which PRs are the next in queue. Can you let me know which ones are ready for review?

**Comment by JackKuei** [comment]
@jtopjian Could you go in this order, Profiles->Clusters->Nodes, and then CRUD. These two are on deck: Clusters Create, Node Create.  
https://github.com/gophercloud/gophercloud/pull/872 https://github.com/gophercloud/gophercloud/pull/873

**Comment by JackKuei** [comment]
Yes. But seems like nodes might be stuck until get a response from the senlin folks regarding data for docker

**Comment by jtopjian** [comment]
Understood.

Are there other resources that will depend on the `Nodes.Data` information? Or is it strictly for informational purposes? If the latter, then the quickest+safest option would be to do `map[string]interface{}` for `Data`.

I'd rather do something like `map[string]interface{}` and follow-up with breaking changes down the road to include a more detailed model of the field rather than merge something with the potential to not unmarshal correctly.

**Comment by jtopjian** [comment]
@JackKuei Thank you for your continued work on this. 

There are a lot of PRs to go through here. Since you and I average 1-2 review requests / review amends per day, it's going to take a good amount of time to get through all of this.

I've recently got a local Senlin environment running. One option we can do is for me to look at these in bulk locally and amend them with any minor style changes myself. If I come across something that requires further discussion, I can merge the prior work and we can discuss it in the PR. That should help accelerate things. 

Thoughts?

**Comment by JackKuei** [comment]
@jtopjian That sounds great! There was alot of manual work for resolving conflicts every time one of the PR gets merged, if you can bulk review them, will definitely speed up the process.
